src/contexts/TodoContext.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { TodoContextTypes } from '../types/TodoContextTypes';
+import { Status } from '../types/Status';
+import { ErrorsMessage } from '../types/ErrorsMessage';
+import { getFilteredTodos } from '../services/getFilteredTodos';
+import { getTodos } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { USER_ID } from '../constants/User';
+
+export const TodoContext = React.createContext<TodoContextTypes>({
+  todos: [],
+  filteredTodos: [],
+  status: Status.All,
+  tempTodo: null,
+  errorMessage: ErrorsMessage.None,
+  updatingTodosIds: [],
+  addTodo: () => {},
+  deleteTodo: () => {},
+  handleSetTempTodo: () => {},
+  handleSetErrorMessage: () => {},
+  handleUpdatingTodosIds: () => {},
+  changeStatus: () => {},
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodoProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [errorMessage, setErrorMessage] = useState<ErrorsMessage>(
+    ErrorsMessage.None,
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
```

Comment:
If you put null to initial state you already have type Todo | null so you can write just Todo in type